In [43]:
import pandas as pd
# pd.set_option('max columns', 20)

# 1. IT Salary Survey EU 2018
Read the CSV file using pandas

In [44]:
df_2018 = pd.read_csv("IT Salary Survey EU 2018.csv")

## 1. Understanding the data
* **Shape** 
    * ```df.shape``` - tuple that tells us the size of our data frame. Row * Column (Like the order of a matrix)
* **Row Analysis** 
    * ```df.head()``` - function that shows the first 5 rows
    * ```df.tail()``` - function that shows the last 5 rows
* **Column Analysis** 
    * ```df.columns``` - used to list all the columns
    * ```df.dtypes``` - determines the data types of all the columns
    * ```df.describe``` - my favorite! Gives small numeric data or statistics about the columns

1. Find out the shape of the data frame

In [45]:
print(df_2018.shape)

(765, 14)


2. As we can see from the shape, we have 14 columns. What are they?

In [46]:
print(df_2018.columns)

Index(['Timestamp', 'Age', 'Gender', 'City', 'Position', 'Years of experience',
       'Your level', 'Current Salary', 'Salary one year ago',
       'Salary two years ago', 'Are you getting any Stock Options?',
       'Main language at work', 'Company size', 'Company type'],
      dtype='object')


3. Determine the data type of the columns

In [47]:
print(df_2018.dtypes)

Timestamp                              object
Age                                   float64
Gender                                 object
City                                   object
Position                               object
Years of experience                   float64
Your level                             object
Current Salary                        float64
Salary one year ago                   float64
Salary two years ago                  float64
Are you getting any Stock Options?     object
Main language at work                  object
Company size                           object
Company type                           object
dtype: object


4. Get insights on the numeric data/statistics about the columns

In [48]:
df_2018.describe()

Age  Years of experience  Current Salary  Salary one year ago  \
count  672.000000           732.000000      750.000000           596.000000   
mean    32.183036             8.548497    68381.765333         62187.278523   
std      5.107268             4.729557    21196.306557         20163.008663   
min     21.000000             0.000000    10300.000000         10001.000000   
25%     29.000000             5.000000    57000.000000         52000.000000   
50%     32.000000             8.000000    65000.000000         60000.000000   
75%     35.000000            11.000000    75000.000000         70000.000000   
max     60.000000            38.000000   200000.000000        200000.000000   

       Salary two years ago  
count            463.000000  
mean           58013.475162  
std            20413.048908  
min            10001.000000  
25%            48000.000000  
50%            56000.000000  
75%            67000.000000  
max           150000.000000

Print the first and last 5 lines of the csv data to know the size of the data we're working with

In [49]:
# First 5
print(df_2018.head())

             Timestamp   Age Gender     City                       Position  \
0  14/12/2018 12:41:33  43.0      M  München                   QA Ingenieur   
1  14/12/2018 12:42:09  33.0      F  München  Senior PHP Magento developer    
2  14/12/2018 12:47:36  32.0      M  München              Software Engineer   
3  14/12/2018 12:50:15  25.0      M  München      Senior Frontend Developer   
4  14/12/2018 12:50:31  39.0      M  München                    UX Designer   

   Years of experience Your level  Current Salary  Salary one year ago  \
0                 11.0     Senior         77000.0              76200.0   
1                  8.0     Senior         65000.0              55000.0   
2                 10.0     Senior         88000.0              73000.0   
3                  6.0     Senior         78000.0              55000.0   
4                 10.0     Senior         69000.0              60000.0   

   Salary two years ago Are you getting any Stock Options?  \
0               68

In [50]:
# Last 5
print(df_2018.tail())

               Timestamp   Age Gender     City               Position  \
760  03/06/2020 20:12:51  40.0      M     Köln  Java Developer junior   
761  28/07/2020 04:03:13   NaN      M     Köln      E.g. C# Developer   
762  28/07/2020 04:03:20   NaN      M     Köln      E.g. C# Developer   
763  26/08/2020 09:06:44   NaN    NaN      NaN                    NaN   
764  21/09/2020 01:47:10  31.0      F  München                    Pm    

     Years of experience Your level  Current Salary  Salary one year ago  \
760                  1.0     Junior         44000.0              40000.0   
761                  1.0     Junior         45000.0              40000.0   
762                  1.0     Junior         45000.0              40000.0   
763                  NaN        NaN             NaN                  NaN   
764                 10.0     Senior        110000.0                  NaN   

     Salary two years ago Are you getting any Stock Options?  \
760               40000.0               

Since our first entry is at index 0, we have 764 entries in total.